Application/Source/Util/ActivityIndicator.swift

```diff
@@ -25,13 +25,13 @@ let successIndicator = AnyObserver<String> { event in
     switch event {
     case .next(let value):
         let hud = JGProgressHUD(style: .extraLight)
-        hud?.indicatorView = JGProgressHUDSuccessIndicatorView()
-        hud?.textLabel?.text = value
-        hud?.backgroundColor = UIColor(white: 0, alpha: 70%)
-        hud?.show(in: UIApplication.shared.keyWindow)
-        hud?.dismiss(afterDelay: 1)
-        hud?.tapOutsideBlock = { hud in
-            hud?.dismiss()
+        hud.indicatorView = JGProgressHUDSuccessIndicatorView()
+        hud.textLabel.text = value
+        hud.backgroundColor = UIColor(white: 0, alpha: 70%)
+        hud.show(in: UIApplication.shared.keyWindow ?? UIWindow())
```

Comment:
You should instead `guard let window` to make sure the `keyWindow` exists, not create your own.